server/server.go

```diff
@@ -191,6 +191,7 @@ func initializeDefaultProject(opts ArgoCDServerOpts) error {
 // NewServer returns a new instance of the Argo CD API server
 func NewServer(ctx context.Context, opts ArgoCDServerOpts) *ArgoCDServer {
 	settingsMgr := settings_util.NewSettingsManager(ctx, opts.KubeClientset, opts.Namespace)
+	settingsMgr.DisableAuth = opts.DisableAuth
```

Comment:
SettingsManager does not really need `DisableAuth` field. Instead please add `disableAuth` field to settings server directly.